app/javascript/components/subject/household_actions/ApplyToHousehold.js

```diff
@@ -0,0 +1,352 @@
+      }
+    );
+  };
+
+  /**
+   * Callback called when child Table component detects a sort change.
+   * @param {Object} sort - Object containing sort field and direction
+   */
+  handleTableSort = sort => {
+    const orderBy = sort.orderBy;
+    let direction = sort.sortDirection;
+    let rowData = _.cloneDeep(this.state.table.rowData);
+    switch (orderBy) {
+      case 'name':
+        rowData = this.sortByName(rowData, direction);
+        break;
+      case 'date_of_birth':
+        rowData = this.sortByDOB(rowData, direction);
+        break;
+      case 'monitoring':
+        // flop the order by to account for the boolean words to be alphabetical
+        direction = direction === 'asc' ? 'desc' : 'asc';
+        rowData = this.sortByBooleanField(rowData, orderBy, direction);
+        break;
+      default:
+        rowData = this.sortByBooleanField(rowData, orderBy, direction);
+    }
+
+    const selectedRows = this.updateSelectedRows(rowData);
+    const disabledRows = this.updateDisabledRows(rowData);
+    this.setState(state => {
+      return {
+        table: { ...state.table, rowData, selectedRows, disabledRows },
+      };
+    });
+  };
+
+  /**
+   * Sorts array of monitorees by name (first name, then last name)
+   * @param {Object[]} patients - array of patient objects
+   * @param {String} direction - direction in which to sort
+   */
+  sortByName = (patients, direction) => {
+    if (direction === 'asc') {
+      patients
+        .sort((a, b) => {
+          return a.first_name.localeCompare(b.first_name);
+        })
+        .sort((a, b) => {
+          return a.last_name.localeCompare(b.last_name);
+        });
+    } else if (direction === 'desc') {
+      patients
+        .sort((a, b) => {
+          return b.first_name.localeCompare(a.first_name);
+        })
+        .sort((a, b) => {
+          return b.last_name.localeCompare(a.last_name);
+        });
+    }
+    return patients;
+  };
+
+  /**
+   * Sorts array of monitorees by date of birth
+   * @param {Object[]} patients - array of patient objects
+   * @param {String} direction - direction in which to sort
+   */
+  sortByDOB = (patients, direction) => {
+    if (direction === 'asc') {
+      patients.sort((a, b) => {
+        return moment(a.date_of_birth).format('YYYYMMDD') - moment(b.date_of_birth).format('YYYYMMDD');
+      });
+    } else {
+      patients.sort((a, b) => {
+        return moment(b.date_of_birth).format('YYYYMMDD') - moment(a.date_of_birth).format('YYYYMMDD');
+      });
+    }
+    return patients;
+  };
+
+  /**
+   * Sorts array of monitorees by a specified boolean field on the patient
+   * @param {Object[]} patients - array of patient objects
+   * @param {String} field - field in which to sort by
+   * @param {String} direction - direction in which to sort
+   */
+  sortByBooleanField = (patients, field, direction) => {
+    if (direction === 'asc') {
+      patients.sort((a, b) => {
+        return a[field] - b[field];
```

Comment:
Github is showing a lot of warnings here for Generic Object Injection Sink because it doesn't know what `field` is. Think you can add a `hasOwnProperty` check to resolve. See more [here](https://stackoverflow.com/a/64207545).

Reply:
added here: https://github.com/SaraAlert/SaraAlert/pull/722/commits/a86b1bf1a275462a4225aff7b5c69bd047cbda6e

Reply:
Warnings are still showing up it looks like

Reply:
Not sure what else to do then besides having separate methods for everything but that's a lot of duplication

Reply:
Given that these fields are determined by the Patient model and not user input, I think it's safe to say that this isn't an actual security issue. Because of that, we can avoid the ESLINT warning by doing something like:
```
 a[`${field}`] - b[`${field}`];
```
Just leave a comment to explain why it's like this. If VSCode doesn't already point out these issues to you inline, you can run locally you can test locally with `eslint app/javascript/components/`.

Reply:
done here: https://github.com/SaraAlert/SaraAlert/pull/722/commits/9f39c348e40e4c9953b49ac6eebb82cdac2474e0